src/foam/dao/PublicUserInfoDAO.java

```diff
@@ -0,0 +1,112 @@
+package foam.dao;
+
+import foam.dao.*;
+import foam.core.FObject;
+import foam.core.PropertyInfo;
+import foam.core.X;
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.dao.Sink;
+import foam.nanos.auth.PublicUserEntity;
+import foam.nanos.auth.User;
+import net.nanopay.tx.model.Transaction;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.logger.Logger;
+
+/**
+ * Populate a model returned from find or select with a small subset of public 
+ * user information.
+ *
+ * Example: A call like
+ *
+ *   new foam.dao.PublicUserInfoDAO(x, "payerId", "payer", ...)
+ *
+ * will look for a property named "payerId" on the object being selected and
+ * look up the user associated with that id. It will then set the given property
+ * ("payer") to the object being returned that contains some public properties
+ * of the associated user.
+ *
+ * Requires both properties to already be defined on the model. For example, on
+ * Transaction.js:
+ *
+ *     ...
+ *   },
+ *   {
+ *    class: 'FObjectProperty',
+ *    of: 'net.nanopay.tx.model.TransactionEntity',
+ *    name: 'payee',
+ *    storageTransient: true
+ *   },
+ *   {
+ *    class: 'Long',
+ *    name: 'payeeId'
+ *   },
+ *   {
+ *     ...
+ */
+public class PublicUserInfoDAO
+  extends ProxyDAO
+{
+  private String idPropertyName_;
```

Comment:
Use protected instead of private to make easier to debug.
Document instance variables.